helpers/test/test_lib_tasks_docker_release.py

```diff
@@ -346,3 +355,419 @@ def test_docker_build_prod_image_with_candidate_user_tag(self) -> None:
+            self.test_version,
+            cache=False,
+            skip_tests=True,
+            fast_tests=False,
+            slow_tests=False,
+            superslow_tests=False,
+            qa_tests=False,
+            target_registries="aws_ecr.ck,dockerhub.causify",
+        )
+        exp = r"""
+        cp -f devops/docker_build/dockerignore.dev .dockerignore
+        docker buildx create \
+            --name multiarch_builder \
+            --driver docker-container \
+            --bootstrap \
+            && \
+            docker buildx use multiarch_builder
+        tar -czh . | DOCKER_BUILDKIT=0 \
+            time \
+            docker buildx build \
+            --no-cache \
+            --push \
+            --platform linux/amd64,linux/arm64 \
+            --build-arg AM_CONTAINER_VERSION=1.0.0 --build-arg INSTALL_DIND=True --build-arg POETRY_MODE=update --build-arg CLEAN_UP_INSTALLATION=True \
+            --tag test.ecr.path/test-image:local-$USER_NAME-1.0.0 \
+            --file devops/docker_build/dev.Dockerfile \
+            -
+        docker pull test.ecr.path/test-image:local-$USER_NAME-1.0.0
+        invoke docker_cmd --stage local --version 1.0.0 --cmd 'cp -f /install/poetry.lock.out /install/pip_list.txt .' --skip-pull
+        cp -f poetry.lock.out ./devops/docker_build/poetry.lock
+        cp -f pip_list.txt ./devops/docker_build/pip_list.txt
+        docker image ls test.ecr.path/test-image:local-$USER_NAME-1.0.0
+        docker buildx imagetools create -t test.ecr.path/test-image:dev-1.0.0 test.ecr.path/test-image:local-$USER_NAME-1.0.0
+        docker buildx imagetools create -t test.ecr.path/test-image:dev test.ecr.path/test-image:local-$USER_NAME-1.0.0
+        docker buildx imagetools create -t causify/helpers:dev-1.0.0 test.ecr.path/test-image:local-$USER_NAME-1.0.0
+        docker buildx imagetools create -t causify/helpers:dev test.ecr.path/test-image:local-$USER_NAME-1.0.0
+        """
+        self._check_docker_command_output(exp, self.mock_run.call_args_list)
+
+
+# #############################################################################
+# Test_docker_rollback_dev_image1
+# #############################################################################
+
+
+class Test_docker_rollback_dev_image1(_DockerFlowTestHelper):
+
+    def test_docker_rollback_dev_image(self) -> None:
+        """
+        Test rolling back a dev Docker image.
+
+        This test verifies that the correct sequence of commands is
+        generated for rolling back a dev Docker image to a specific
+        version.
+        """
+        # Call tested function.
+        hltadore.docker_rollback_dev_image(
+            self.mock_ctx,
+            self.test_version,
+            push_to_repo=True,
+        )
+        exp = r"""
+        docker pull test.ecr.path/test-image:dev-1.0.0
+        docker tag test.ecr.path/test-image:dev-1.0.0 test.ecr.path/test-image:dev
+        docker push test.ecr.path/test-image:dev-1.0.0
```

Comment:
Kind of weird, isn't it? 
Pull the versioned dev image and then push it back.
It might be worth to add a TODO to the `docker_push_dev_image()`. 
Your call.